tempesta_fw/sock_srv.c

```diff
@@ -1791,9 +1791,30 @@ tfw_cfgop_grace_time(TfwCfgSpec *cs, TfwCfgEntry *ce)
 	return tfw_cfgop_intval(cs, ce, &tfw_cfg_grace_time_reconfig);
 }
 
+/* Forward declaration */
+static TfwMod tfw_sock_srv_mod;
+
 static int
 tfw_sock_srv_cfgstart(void)
 {
+	TfwCfgSpec *cfg_spec;
+	TfwMod srv_mod;
+	struct list_head mod_list;
+
+	/*
+	 * Two new server group is created there. It's almost like call
+	 * 'srg_group` directive, so force to cleanup if configuration
+	 * parsing has failed and no real 'srv_group' directives hve met.
+	 */
+	srv_mod = tfw_sock_srv_mod;
+	INIT_LIST_HEAD(&srv_mod.list);
+	INIT_LIST_HEAD(&mod_list);
+	list_add(&srv_mod.list, &mod_list);
+	cfg_spec = tfw_cfg_spec_find(srv_mod.specs, "srv_group");
+	BUG_ON(!cfg_spec);
+	cfg_spec->__called_now = true;
+	cfg_spec->__called_cfg = true;
```

Comment:
Frankly, I believe that these are supposed to be flags that are *internal* to `cfg.c`. If this is the only way to make things work correctly in this case (a necessary evil), then I think it's best to wrap setting of these flags into a function with a meaningful name specifically for a case like this.